cmd/geth/main.go

```diff
@@ -417,6 +428,35 @@ func attach(ctx *cli.Context) {
 	}
 }
 
+func initGenesis(ctx *cli.Context) {
+	genesisPath := ctx.Args().First()
+	if len(genesisPath) == 0 {
+		utils.Fatalf("must supply path to genesis JSON file")
+	}
+
+	var (
+		datadir = utils.MustMakeDataDir(ctx)
+		cache   = ctx.GlobalInt(utils.CacheFlag.Name)
+		handles = utils.MakeDatabaseHandles()
+	)
+
+	chainDb, err := ethdb.NewLDBDatabase(filepath.Join(datadir, "chaindata"), cache, handles)
+	if err != nil {
+		utils.Fatalf("could not open database: %v", err)
+	}
+
+	genesisFile, err := os.Open(genesisPath)
+	if err != nil {
+		utils.Fatalf("failed to read genesis file: %v", err)
+	}
+
+	block, err := core.WriteGenesisBlock(chainDb, genesisFile)
+	if err != nil {
+		utils.Fatalf("failed to write genesis block: %v", err)
+	}
+	glog.V(logger.Info).Infof("successfully wrote genesis block: %x", block.Hash())
```

Comment:
Will add the return value when we need it. I'm not a fan of changing code for a maybe we'll need in the future; if we do we can always change it later :-)

I think this log line (successfully wrote ...) should be moved up to `WriteGenesisBlock` where we can also add the additional log message that mentions the rule set write.